KBQA/data-generator/dataset.py

```diff
@@ -11,8 +12,8 @@ def __init__(
         self,
         question: str,
         sparql: str = "",
-        answers: List = list(),
-        triples: List = list(),
+        answers: Union[list, None] = None,
```

Comment:
```suggestion
        answers: Optional[List] = None,
```
Optional is a shorthand notation and List from typing should be used.